src/mantine-hooks/src/use-list-state/use-list-state.ts

```diff
@@ -77,7 +77,10 @@ export function useListState<T>(initialValue: T[] = []): UseListState<T> {
   const applyWhere = (
     condition: (item: T, index?: number) => boolean,
```

Comment:
Here the index should not be optional, the type should be:
`condition: (item: T, index: number) => boolean,`

Same with type definition on line 12